src/LaunchDarkly.EventSource/EventParser.cs

```diff
@@ -9,119 +10,87 @@ namespace LaunchDarkly.EventSource
     /// </summary>
     internal static class EventParser
     {
-
-        /// <summary>
-        /// Determines if the specified value is a Comment in a Server Sent Event message.
-        /// </summary>
-        /// <param name="value">A string value.</param>
-        /// <returns>
-        ///   <c>true</c> if the specified value is a comment; otherwise, <c>false</c>.
-        /// </returns>
-        public static bool IsComment(string value)
-        {
-            if (string.IsNullOrWhiteSpace(value)) return false;
-
-            return value.StartsWith(":", StringComparison.OrdinalIgnoreCase);
-        }
-
-        /// <summary>
-        /// Determines if the specified value is a data field in a Server Sent Event message.
-        /// </summary>
-        /// <param name="value">The value.</param>
-        /// <returns>
-        ///   <c>true</c> if the specified value is a data field; otherwise, <c>false</c>.
-        /// </returns>
-        public static bool IsDataFieldName(string value)
-        {
-            return Constants.DataField.Equals(value, StringComparison.OrdinalIgnoreCase);
```

Comment:
The logic for all of these `Is...FieldName` methods was incorrect: they were doing a case-insensitive comparison, but the [SSE spec](https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation) says "Field names must be compared literally, with no case folding performed." I've changed this so it just does regular string comparisons in the `Is...` properties of `Result`.

That's a breaking change, sort of, in that it means EventSource will fail to process data that it would've previously accepted. But since that is by definition not valid stream data at all, it's more of a bug fix.